gyp/gyp_main.py

```diff
@@ -6,10 +6,38 @@
 
 import os
 import sys
+import subprocess
+
+# Below IsCygwin() function copied from pylib/gyp/common.py
+def IsCygwin():        
```

Comment:
@cclauss I wish (and sure there is) a way to put this function in a common place so that `common.py` can re-use  from there was as well. Right now this code is duplicated in `common.py`. I'm new to Python so did not know off the top of my head of a way to consolidate. Can spend additional time later today looking into it deeper if need be.

Reply:
__from common import IsCygwin__

Reply:
Alas, did not work:

```
Traceback (most recent call last):
  File "C:\Users\v094303\AppData\Roaming\npm\node_modules\node-gyp\gyp\gyp_main.py", line 10, in <module>
    from gyp.common import IsCygwin
ImportError: No module named gyp.common
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\Users\v094303\AppData\Roaming\npm\node_modules\node-gyp\lib\configure.js:328:16)
gyp ERR! stack     at ChildProcess.emit (events.js:197:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:254:12)
gyp ERR! System Windows_NT 10.0.16299
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\v094303\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\git\ecom\git-repo-inspector\node_modules\@data-leakage-protection\signatures\node_modules\re2
gyp ERR! node -v v11.9.0
gyp ERR! node-gyp -v v5.0.0
gyp ERR! not ok
```

I think because the call to `sys.path.insert(0, os.path.join(os.path.dirname(path), 'pylib'))` needs to have happened before attempting to `import gyp.common`, but that call relies on the Cygwin check to properly format the path to insert. I'm fine leaving the duplicate code in pace for now.

```
# Make sure we're using the version of pylib in this repo, not one installed
# elsewhere on the system. Also convert to Unix style path on Cygwin systems,
# else the 'gyp' library will not be found
path = UnixifyPath(sys.argv[0])
sys.path.insert(0, os.path.join(os.path.dirname(path), 'pylib'))
import gyp
```

Reply:
Let's leave the duplicated functions in for now and open a new issue for me to remove the duplication in a future PR.

Reply:
Will do.